### PR TITLE
SWIFT-135 Make a contributor guide

### PR DESCRIPTION
**Pull request by kmahar**
I moved the building and testing instructions out into a separate file, and added more information on things like linting, workflow, etc.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -145,39 +141,12 @@ var doc: Document = ["a": 1, "b": 2, "c": 3]
 print(doc) // prints `{"a" : 1, "b" : 2, "c" : 3}`
 print(doc["a"] ?? "") // prints `1`
 
+
```
Comment (jmikola) [inline README.md]: Any reason for two line breaks here?

File: development.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
+# Swift Driver Development Guide
+
+## Index
+* [Things to install](#things-to-install) 
+* [The code](#the-code)
+* [Building](#building)
+* [Running tests](#running-tests)
```
Comment (jmikola) [inline development.md]: Title capitalization:

 * Things to Install
 * The Code
 * Running Tests

File: development.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+**NOTE**: Several of the tests require a mongod instance to be running on the default host/port, `localhost:27017`.
+
+Additionally, please note that each benchmark test runs for a minimum of 1 minute and therefore **the entire benchmark suite will take around 20-30 minutes to complete**.
+
+You can run tests from Xcode as usual. If you prefer to test from the command line, keep reading.
+
+### From the command line 
+Tests can be run from the command line with `make test`. By default, this will run all the tests excluding the benchmarks.
+
+To only run particular tests, use the `FILTER` argument, which is passed as the `filter` argument to `swift test`. This will run test cases with names matching a regular expression, formatted as follows: `<test-target>.<test-case>` or `<test-target>.<test-case>/<test>`.
```
Comment (jmikola) [inline development.md]: > use the `FILTER` argument

Should this be "`FILTER` environment variable"? It's an argument/parameter to `swift test`, but in the example below it does look like an environment var.

File: development.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+For example, `make test FILTER=ClientTests` will run `MongoSwiftTests.ClientTests/*`. Or, `make test FILTER=testInsertOne` will only run `MongoSwiftTests.CollectionTests/testInsertOne`. 
+
+To run all of the benchmarks, use `make benchmark` (equivalent to `FILTER=MongoSwiftBenchmarks`). To run a particular benchmark, use the `FILTER` argument to specify the name. To have the benchmark results all printed out at the end, run with `make benchmark | python Tests/MongoSwiftBenchmarks/benchmark.py`.
```
Comment (jmikola) [inline development.md]: Another reference to "`FILTER` argument" that may be changed (if the above observation is correct).

File: development.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+## Writing and Generating Documentation
+We document new code as we write it. We use C-style comments for documentation longer than 3 lines, and triple-slash (`///`) for shorter documentation.
```
Comment (jmikola) [inline development.md]: Are all C-style comments documentation, or just doc blocks starting with `/**` ?

File: development.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+### Sublime Text setup
+If you use Sublime Text, you can get linting violations shown in the editor by installing the packages [SublimeLinter](https://packagecontrol.io/packages/SublimeLinter) and [SublimeLinter-contrib-swiftlint](https://packagecontrol.io/packages/SublimeLinter-contrib-swiftlint). 
+
+## Workflow
+1. Create a feature branch, named by the corresponding JIRA ticket if exists; for example, `SWIFT-30`. 
+2. Do your work on the branch.
+3. Open a pull request on the repository. Make sure you have rebased your branch onto the latest commits on `master`. 
+
+Once you get the required approvals and your code passes all tests….
```
Comment (jmikola) [inline development.md]: Should "…." just be a colon?

File: development.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+* [Building](#building)
+* [Running tests](#running-tests)
+* [Writing and Generating Documentation](#writing-and-generating-documentation)
+* [Linting](#linting)
+* [Workflow](#workflow)
+* [Resources](#resources)
+
+## Things to install
+* [swiftenv](https://swiftenv.fuller.li/en/latest/installation.html): a command-line tool that allows easy installation of and switching between versions of Swift.
+	* Use this to install Swift 4.0 if you don’t have it already.
```
Comment (jmikola) [inline development.md]: Just an observation, but this file has some inconsistent apostrophes (`’` and `'`). May be preferable to stick to the basic ASCII character for a plain text file like Markdown.

File: development.md [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+7. Squash all commits into a single, descriptive commit method, formatted as: `TICKET-NUMBER: Description of changes`. For example, `SWIFT-30: Implement WriteConcern type`. 
+8. Merge it, or if you don’t have permissions, ask someone to merge it for you.
+
+## Resources
+
+### Swift
+* [Swift Language Guide](https://docs.swift.org/swift-book/LanguageGuide/TheBasics.html)
+* [Swift Standard Library docs](https://developer.apple.com/documentation/swift)
+
+### MongoDB and drivers
```
Comment (jmikola) [inline development.md]: "and Drivers"

File: development.md [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+We document new code as we write it. We use C-style comments for documentation longer than 3 lines, and triple-slash (`///`) for shorter documentation. 
+Comments that are _not_ documentation should use two slashes (`//`).
+
+Our documentation site is automatically generated from the source code using [jazzy](https://github.com/realm/jazzy#installation). 
+To regenerate the files after making changes, run `make documentation` from the project's root directory. You can then inspect the changes to the site by opening the files in `/docs` in your web browser.
+
+## Linting
+We use [swiftlint](https://github.com/realm/SwiftLint#using-homebrew) for linting. You can see our configuration in the `.swiftlint.yml` file in the project’s root directory.  Run `swiftlint` in the `/Sources` directory to lint all of our files. Running `swiftlint autocorrect` will correct some types of violations.
+
+### Sublime Text setup
```
Comment (jmikola) [inline development.md]: "Setup"

File: development.md [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+Why is this necessary? The project requires a customized "copy resources" build phase to include various test `.json` files. By default, this phase is not included when you run `swift package generate-xcodeproj`. So `make project` first generates the project, and then uses `xcodeproj` to manually add the files to the appropriate targets (see `add_json_files.rb`). 
+
+## Running Tests
+**NOTE**: Several of the tests require a mongod instance to be running on the default host/port, `localhost:27017`.
+
+Additionally, please note that each benchmark test runs for a minimum of 1 minute and therefore **the entire benchmark suite will take around 20-30 minutes to complete**.
+
+You can run tests from Xcode as usual. If you prefer to test from the command line, keep reading.
+
+### From the command line
```
Comment (jmikola) [inline development.md]: "From the Command Line"

File: development.md [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+* [jazzy](https://github.com/realm/jazzy#installation): the tool we use to generate documentation.
+* [swiftlint](https://github.com/realm/SwiftLint#using-homebrew): the Swift linter we use. 
+* [libmongoc](http://mongoc.org/libmongoc/current/api.html): the MongoDB C driver, which this library wraps. See the installation instructions provided in our [README](README.md#first-install-the-mongodb-c-driver) or on the [libmongoc docs](http://mongoc.org/libmongoc/current/installing.html). 
+
+## The code
+You should clone this repository, as well as the [MongoDB Driver specifications](https://github.com/mongodb/specifications). 
+Since this library wraps the MongoDB C Driver, we also recommend cloning [mongodb/mongo-c-driver](https://github.com/mongodb/mongo-c-driver) so you have the source code and documentation easily accessible. 
+
+## Building 
+### From the command line
```
Comment (jmikola) [inline development.md]: "From the Command Line"

File: README.md (hunk 1 above)
Comment (kmahar) [reply README.md]: nope, removed

File: development.md (hunk 2 above)
Comment (kmahar) [reply development.md]: 👍

File: development.md (hunk 7 above)
Comment (kmahar) [reply development.md]: yeah, agreed, I think this was from copy-pasting. converted all to `'`

File: development.md (hunk 11 above)
Comment (kmahar) [reply development.md]: 👍

File: development.md (hunk 10 above)
Comment (kmahar) [reply development.md]: 👍

File: development.md (hunk 9 above)
Comment (kmahar) [reply development.md]: 👍

File: development.md (hunk 6 above)
Comment (kmahar) [reply development.md]: sure

File: development.md (hunk 8 above)
Comment (kmahar) [reply development.md]: 👍

File: development.md (hunk 5 above)
Comment (kmahar) [reply development.md]: just `/**`. updated to clarify

File: development.md (hunk 3 above)
Comment (kmahar) [reply development.md]: updated

**Pull request merged**